### ISSUE TITLE
Remote writes not sending `job` label after the first few writes

### ISSUE THREAD

**Issue by tao-lu** [prometheus/prometheus#3080]
[NOTICE]: <> (If your question is around usage and not a bug in Prometheus please use: https://groups.google.com/forum/#!forum/prometheus-users)

**What did you do?**
Configured prometheus to write to a remote endpoint

**What did you expect to see?**
A constant flow of requests to that endpoint including a `job` label

**What did you see instead? Under which circumstances?**
The requests following the first few requests did not include `job`. The change occurs fairly quickly in a fresh/plain install where the only configuration is setting a remote write URL.

**Environment**
Mac OSX Sierra 10.12.6 and Centos RHEL 7

* System information:

	Darwin 16.7.0 x86_64 and Linux 3.10.0-514.26.2.el7.YAHOO.20170707.8.x86_64 x86_64

* Prometheus version: 

prometheus, version 2.0.0-beta.1 (branch: HEAD, revision: 4dcb465029f39a2a3030333a81753f8a40ec90ce)
  build user:       root@d3f9974fac5a
  build date:       20170811-12:16:25
  go version:       go1.8.3

* Prometheus configuration file:
```
# my global config
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

  # Attach these labels to any time series or alerts when communicating with
  # external systems (federation, remote storage, Alertmanager).
  external_labels:
      monitor: 'codelab-monitor'

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  # - "first.rules"
  # - "second.rules"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'

    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.

    static_configs:
      - targets: ['localhost:9090']
remote_write:
  - url: http://localhost:1337/write
```

**Comment by tao-lu** [comment]
It seems to be the case that when a metric is cached, the "rule labels" are not appended, i.e. the behaviour that `ruleLabelsAppender` offers does not occur in the case where the metric is cached (by cached, I mean when the conditional statement on line 776 of `retrieval/scrape.go` is true).

**Comment by brian-brazil** [comment]
@tomwilkie

**Comment by tao-lu** [comment]
A possibly related issue is the caching dropping certain metrics and not just certain labels. When I disabled caching (by setting `ok = false`), I noticed a certain metric that was only showing up in the first few writes now showing up consistently among the writes.

**Comment by krasi-georgiev** [comment]
DIBS

**Comment by fabxc** [comment]
This should be fixed via #3151. Unfortunately, it didn't go into beta.3 by accident. We should probably cut beta.4 for that.

**Comment by Gouthamve** [comment]
Can this be closed?
